File: Data Structures & Algorithms/two-integer-sum/submission-0.cs
public class Solution {
     public int[] TwoSum(int[] nums, int target) {
         int[] final = new int[2];
        Dictionary<int,int> ints = new Dictionary<int, int>(); //number and diff with target
        for(int i = 0 ; i<nums.Count() ; i++)
        {
            int diff = target - nums[i];
            if (ints.ContainsKey(diff))
            {
               Console.WriteLine("ints does not contain key" + diff);
                final[0] = ints[diff];
                final[1] = i;
                return final;
               
            }
            else
            {
             Console.WriteLine("current" + nums[i] + " target " + target);
                ints.Add(nums[i], i);
            }
                
        }
        return null;
        }
    }

File: Data Structures & Algorithms/first-missing-positive/submission-0.swift
class Solution {
    func firstMissingPositive(_ nums: inout [Int]) -> Int {
        
        let n = nums.count
        
        for i in 0..<n {
            while nums[i] > 0 &&
                  nums[i] <= n &&
                  nums[i] != nums[nums[i] - 1] {
                
                nums.swapAt(i, nums[i] - 1)
            }
        }
        
        for i in 0..<n {
            if nums[i] != i + 1 {
                return i + 1
            }
        }
        
        return n + 1
    }
}